### ISSUE TITLE
Accessibility check @emplums

### ISSUE THREAD

**Issue by emplums** [primer/components#325]
- [ ] aria labels / other a11y attributes should be able to be passed down via props for appropriate components
   - [x] Button
   - [x] TextInput
   - [ ] Octicon
   - [ ] CircleOcticon
   - [x] CircleBadge
- [x] add guidelines for using a link vs button in docs
- [x] double check that jsx-a11y is running on lint
- [ ] run Chrome a11y audit on docs
- [x] alt tags can be passed down for any `img` tags, documentation encourages users to do so
- [x] add blurb to Heading docs about appropriate heading structure & a11y implications
- [x] make sure API of TextInput makes it easy to apply a label, make sure docs instruct to do so
- [x] make sure `outline: none` isn't being used anywhere
